.github/workflows/docker.yaml

```diff
@@ -0,0 +1,91 @@
+# This workflow builds docker images on 'master' and for all release tags. The
+# 'latest' docker tag on the registry will always point to the latest pushed
+# version, likely the one built from 'master', so referring to the versioned
+# images is suggested.
+name: Docker
+
+# Limit concurrent runs of this workflow within a single PR
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches: [ "master" ]
+    tags: [ "*.*.*" ]
+  workflow_dispatch:
+    inputs:
+      ref_name:
+        type: string
+        description: 'Point-in-time to build the custom docker images'
+        required: true
+        default: "master"
+
+permissions:
+  packages: write
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+    - name: 📥 Checkout repository
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.inputs.ref_name || '' }}
+
+    - name: 🐳 Login to GitHub Container Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: ❄ Prepare nix
+      uses: cachix/install-nix-action@v30
+      with:
+        extra_nix_config: |
+          accept-flake-config = true
+          log-lines = 1000
+
+    - name: ❄ Cachix cache of nix derivations
+      uses: cachix/cachix-action@v15
+      with:
+        name: cardano-scaling
+        authToken: '${{ secrets.CACHIX_CARDANO_SCALING_AUTH_TOKEN }}'
+
+    - name: 🔨 Build image using nix
+      run: |
```

Comment:
Could: keep bash script in files

Reply:
The script would not work as-is as it requires the `${{github ..}}` variable replacing and require >5 arguments to be invokable. Not convinced that this + the indirection of needing to look into another file is worth it. Also, this is just what we have in `hydra` (even simplified).